core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -281,7 +302,9 @@ class ReplicaFetcherThread(name: String,
   override def buildLeaderEpochRequest(allPartitions: Seq[(TopicPartition, PartitionFetchState)]): Map[TopicPartition, Int] = {
     val result = allPartitions
       .filter { case (_, state) => state.isTruncatingLog }
-      .map { case (tp, _) => tp -> epochCache(tp).latestEpoch }.toMap
+      .map { case (tp, _) => tp -> epochCache(tp) }
```

Comment:
Should we call handlePartitionsWithErrors() on those partitions w/o leaderEpoch?

Reply:
Good point. I have updated the patch as suggested.